l1-contracts/test/validator-selection/ValidatorSelection.t.sol

```diff
@@ -11,132 +11,37 @@ import {Signature} from "@aztec/core/libraries/crypto/SignatureLib.sol";
-  mapping(address proposer => address attester) internal proposerToAttester;
-  mapping(address => bool) internal _seenValidators;
-  mapping(address => bool) internal _seenCommittee;
-
-  /**
-   * @notice  Set up the contracts needed for the tests with time aligned to the provided block name
-   */
-  modifier setup(uint256 _validatorCount) {
-    string memory _name = "mixed_block_1";
-    {
-      DecoderBase.Full memory full = load(_name);
-      uint256 slotNumber = full.block.decodedHeader.slotNumber;
-      uint256 initialTime =
-        full.block.decodedHeader.timestamp - slotNumber * TestConstants.AZTEC_SLOT_DURATION;
-
-      timeCheater = new TimeCheater(
-        address(rollup),
-        initialTime,
-        TestConstants.AZTEC_SLOT_DURATION,
-        TestConstants.AZTEC_EPOCH_DURATION
-      );
-      vm.warp(initialTime);
-    }
-
-    CheatDepositArgs[] memory initialValidators = new CheatDepositArgs[](_validatorCount);
-
-    for (uint256 i = 1; i < _validatorCount + 1; i++) {
-      uint256 attesterPrivateKey = uint256(keccak256(abi.encode("attester", i)));
-      address attester = vm.addr(attesterPrivateKey);
-      attesterPrivateKeys[attester] = attesterPrivateKey;
-      uint256 proposerPrivateKey = uint256(keccak256(abi.encode("proposer", i)));
-      address proposer = vm.addr(proposerPrivateKey);
-      proposerPrivateKeys[proposer] = proposerPrivateKey;
-
-      proposerToAttester[proposer] = attester;
-
-      initialValidators[i - 1] = CheatDepositArgs({
-        attester: attester,
-        proposer: proposer,
-        withdrawer: address(this),
-        amount: TestConstants.AZTEC_MINIMUM_STAKE
-      });
-    }
-
-    testERC20 = new TestERC20("test", "TEST", address(this));
-    Registry registry = new Registry(address(this), testERC20);
-    rewardDistributor = RewardDistributor(address(registry.getRewardDistributor()));
-    rollup = new Rollup({
-      _feeAsset: testERC20,
-      _rewardDistributor: rewardDistributor,
-      _stakingAsset: testERC20,
-      _governance: address(this),
-      _genesisState: TestConstants.getGenesisState(),
-      _config: TestConstants.getRollupConfigInput()
-    });
-    slasher = Slasher(rollup.getSlasher());
-    slashFactory = new SlashFactory(IValidatorSelection(address(rollup)));
-
-    testERC20.mint(address(this), TestConstants.AZTEC_MINIMUM_STAKE * _validatorCount);
-    testERC20.approve(address(rollup), TestConstants.AZTEC_MINIMUM_STAKE * _validatorCount);
-    rollup.cheat__InitialiseValidatorSet(initialValidators);
-
-    inbox = Inbox(address(rollup.getInbox()));
-    outbox = Outbox(address(rollup.getOutbox()));
-
-    merkleTestUtil = new MerkleTestUtil();
-
-    // Progress into the next epoch for changes to take effect
-    timeCheater.cheat__progressEpoch();
-    _;
-  }
-
-  function testInitialCommitteeMatch() public setup(4) {
+  function testInitialCommitteeMatch() public setup(4) progressEpoch {
```

Comment:
If we apply the `progressEpoch` to each one, whould it not make sense to apply it to the setup so there is one less thing to think about?